gprofiler/system_metrics.py

```diff
@@ -0,0 +1,108 @@
+import statistics
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from threading import Event, RLock, Thread
+from typing import List, Optional
+
+import psutil
+
+from gprofiler.exceptions import ThreadStopTimeoutError
+
+DEFAULT_POLLING_INTERVAL_SECONDS = 5
+STOP_TIMEOUT_SECONDS = 30
```

Comment:
Too high. This blocks after the user has requested us to stop. Not that I think this will ever trigger, but anyway, bring it lower (2 seconds are more than enough)